### PR TITLE
docs - dashboard subscriptions - closes #14075

### LINKED ISSUE #14075: Document dashboard subscriptions
- [Product doc](https://docs.google.com/document/d/1CmXQKvz30iFGRfA0HFzF9f2KI9mFKo8wzWO6YjP_2SE/edit)
- [Implementation issue](https://github.com/metabase/metabase/issues/13671)

Dashboard subscriptions allow you to send the contents of a dashboard via email or slack regularly on a schedule, just like pulses. In fact, in 0.38 we'll be marking pulses as deprecated, and will then remove them in a future release along with a way to help users migrate any pulses they have left over to dashboard subscriptions.

The emails or slack messages that these subscriptions send are essentially identical to pulses. They will include all cards that are in the dashboard at the time the subscription gets sent, excluding text/markdown cards. So if you remove or add cards from/to a dashboard, those changes will be reflected in the subscription.

The EE will include added features to allow setting filter values on the dashboard on a per-subscription basis, and to more granularly control who can set up and receive subscriptions.

To try out this feature:

- Set up email or slack locally (or just go do this on our internal instance)
- Go to a dashboard
- Click on the sharing and embedding button in the top-right and select the create subscription option
- Pick email or slack
- Choose where you want to send the dashboard, and optionally include some or all of the questions' results as attachments
- Click the send-now button, or click Done and then go make a cup of coffee and wait. :)

### PR DESCRIPTION
**Pull request by jeff-bruemmer**
Draft PR for 0.38 docs covering dashboard subscriptions. Consider the screenshots placeholders, as we may need to retake them if the UI changes. I didn't take a shot of the Slack example, as the Slack message still had the word "pulse" in it.

Added a section in the pulses doc that telegraphs the deprecation, and directs people to dashboard subscriptions. In a future release that deprecates pulses, we'll need to redirect inbound links to the pulses page to this new dashboard subscriptions page.

### CODE REVIEW

File: docs/users-guide/dashboard-subscriptions.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+![Dashboard subscription email options](images/dashboard-subscriptions/email-options.png)
+
+### Email subscription options
+
+For emails, we can:
+
+- **Add subscribers**. We can add subscribers to the dashboard by adding their email addresses.
+- **Determine frequency and timing**. Determine how often Metabase sends the dashboard (daily, weekly, or monthly) and what time of day Metabase sends the dashboard.
+- **Attach results**. Whether Metabase should also attach results to the email (which will include up 2000 rows of data). Note that Metabase will automatically attach results as a table for questions that do not translate
```
Comment (mazameli) [inline docs/users-guide/dashboard-subscriptions.md:29]: Looks like this is an unfinished paragraph.

File: docs/users-guide/dashboard-subscriptions.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+- **Skip updates without results**. If there are no results, we can tell Metabase to skip sending the dashboard at all.
+- **Immediate gratification:** Or send an email right now, which is great for sending a test email.
```
Comment (mazameli) [inline docs/users-guide/dashboard-subscriptions.md:31]: I appreciate the humor but don't know if this is clear, and leaves me with questions about the actual functionality of the send-now button, e.g., does this test email get sent to me, to all the recipients, or…?

File: docs/users-guide/dashboard-subscriptions.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+![Set up a dashboard subscription with email or slack](images/dashboard-subscriptions/email-or-slack.png)
+
+Let's say we want to email a dashboard. We'll click on the **Email it** option in the sidebar, and Metabase will give us some options:
+
+![Dashboard subscription email options](images/dashboard-subscriptions/email-options.png)
+
+### Email subscription options
+
+For emails, we can:
```
Comment (mazameli) [inline docs/users-guide/dashboard-subscriptions.md:25]: I'd advocate for listing the options in the order they appear. I think the send-now button's bullet and the attach results bullet are in the wrong positions.

File: docs/users-guide/dashboard-subscriptions.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+- **Attach results**. Whether Metabase should also attach results to the email (which will include up 2000 rows of data). Note that Metabase will automatically attach results as a table for questions that do not translate
+- **Skip updates without results**. If there are no results, we can tell Metabase to skip sending the dashboard at all.
+- **Immediate gratification:** Or send an email right now, which is great for sending a test email.
+
+### Email example
+
+![Example dashboard subscription email](images/dashboard-subscriptions/example-email.png)
+
+You'll notice in the email that Metabase excludes any text cards on the dashboard, and that the charts look different - Metabase reformats the charts to make them more legible in email. Additionally, tables that exceed either 10 columns or 20 rows get the rest of their results, up to 2000 rows, included as an attachment. As a bonus, Metabase will toss in a random data quotation, like this one from the mathematician Andrejs Dunkels: "It's easy to lie with statistics. It's hard to tell the truth without statistics."
```
Comment (mazameli) [inline docs/users-guide/dashboard-subscriptions.md:37]: Dunno that we need to call out the quotation bit. Additionally, once we get the click/linking behavior dialed in, that's something we might want to call out — "Clicking on a question in the email will take you to the dashboard."

```suggestion
You'll notice in the email that Metabase excludes any text cards on the dashboard, and that the charts look different - Metabase reformats the charts to make them more legible in email. Additionally, tables that exceed either 10 columns or 20 rows get the rest of their results, up to 2000 rows, included as an attachment.
```

File: docs/users-guide/dashboard-subscriptions.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+### Slack subscription options
+
+For Slack subscriptions, you can set up a subscription for a single person using their Slack username, or for a channel (like #general).
```
Comment (mazameli) [inline docs/users-guide/dashboard-subscriptions.md:41]: I think you've got the importance reversed here — seems like the more common usecase is sending a slack subscription to a channel. E.g., "You can also send a subscription to yourself or another user in Slack."

File: docs/users-guide/dashboard-subscriptions.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+- **Skip updates without results**. If there are no results, we can tell Metabase to skip sending the dashboard at all.
+- **Immediate gratification:** Or send an email right now, which is great for sending a test email.
+
+### Email example
+
+![Example dashboard subscription email](images/dashboard-subscriptions/example-email.png)
+
+You'll notice in the email that Metabase excludes any text cards on the dashboard, and that the charts look different - Metabase reformats the charts to make them more legible in email. Additionally, tables that exceed either 10 columns or 20 rows get the rest of their results, up to 2000 rows, included as an attachment. As a bonus, Metabase will toss in a random data quotation, like this one from the mathematician Andrejs Dunkels: "It's easy to lie with statistics. It's hard to tell the truth without statistics."
+
+### Slack subscription options
```
Comment (mazameli) [inline docs/users-guide/dashboard-subscriptions.md:39]: Slack subscriptions currently also have an additional schedule option to send hourly I believe.

File: docs/users-guide/dashboard-subscriptions.md (hunk 1 above)
Comment (mazameli) [reply docs/users-guide/dashboard-subscriptions.md:29]: You can also choose between csv or xlsx for your attachments' file format.

**Review by mazameli — COMMENTED** [review]
Just some general comments and suggestions. This is shaping up well I think.

**Comment by mazameli** [comment]
One thing that occurred to me that I'll need to verify — it could be a source of confusion if dashboard filters don't affect the cards when they get sent in the subscription, even if a filter has a default value. I haven't tested to be sure exactly what happens, but I'm assuming that if you e.g. have a dashboard with Revenue, and you've set your dashboard's state filter to `California`, the subscription is going to send the _unfiltered_ Revenue, not the CA revenue. Setting the filter values per subscription is a planned EE feature, though.

File: docs/users-guide/10-pulses.md [hunk 7]
```diff
@@ -1,4 +1,8 @@
-## Sharing updates with pulses
+## Dashboard subscriptions will replace Pulses in a future release
+
+[Dashboard subscriptions](dashboard-subscriptions) will be replacing Pulses in a future release of Metabase. Dashboard subscriptions allow you to send the contents of a dashboard on a schedule, without having to manually add each question (which Pulses require you to do). When we remove Pulses, we'll provide help for migrating your remaining Pulses to dashboard subscriptions. So if you're thinking of setting up a Pulse, try out a dashboard subscription instead.
```
Comment (tsmacdonald) [inline docs/users-guide/10-pulses.md:3]: It seems a little odd that "Pulse" is capitalized and "dashboard subscription" is not.

**Review by tsmacdonald — APPROVED** [review]
I don't think my :heavy_check_mark: means much here, but it looks correct to me!

> Consider the screenshots placeholders, as we may need to retake them if the UI changes. I didn't take a shot of the Slack example, as the Slack message still had the word "pulse" in it.

The screenshots are up to date. And Slack doesn't have "Pulse" in it any more, so screenshot away!

File: docs/users-guide/dashboard-subscriptions.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
+## Dashboard subscriptions
+
+Dashboard subscriptions are a great way to keep you and your team up to date on the data that matters most. They allow you to send all of the questions on a dashboard via email or Slack. If your Metabase has email or Slack set up, all you need to do is create a dashboard, add subscribers to it, and tell Metabase how often you'd like the send out an update. You can set up as many subscriptions to a dashboard as you like, and if you make any changes to the dashboard, Metabase will update the subscription the next time they're delivered.
```
[the remark this answers is not in the document]
Comment (tsmacdonald) [reply docs/users-guide/dashboard-subscriptions.md:3]: (or "the subscription the next time **it's** delivered")

File: docs/users-guide/dashboard-subscriptions.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+For emails, we can:
+
+- **Add subscribers**. Add email addresses to register subscribers.
+- **Determine frequency and timing**. Tell Metabase how often it should send the dashboard (daily, weekly, or monthly), and what time of day to send the dashboard.
+- **Send email now** sends an email to all subscribers.
+- **Skip updates without results**. If there are no results, we can tell Metabase to skip sending the email.
+- **Attach results**. Tell Metabase if it should also attach results to the email (which will include up to 2000 rows of data). You can choose between CSV and XLSX file formats.
+
+If you've added filters to your dashboard and set default values for those filters, the subscriptions will run the dashboard with those default values applied, filtering results for all questions with fields connected to those filters. To learn more, check out [dashboard filters](https://www.metabase.com/docs/latest/users-guide/08-dashboard-filters.html).
```
Comment (mazameli) [inline docs/users-guide/dashboard-subscriptions.md:33]:
```suggestion
If you've added filters to your dashboard and set default values for those filters, those default values will be applied to your subscriptions, filtering the results of all questions that are connected to those filters when the subscriptions are sent. To learn more, check out [dashboard filters](https://www.metabase.com/docs/latest/users-guide/08-dashboard-filters.html).
```

**Review by mazameli — APPROVED** [review]

**Pull request merged**